v3/src/components/web-view/web-view-url-modal.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  Button, FormControl, Input, ModalBody, ModalFooter, ModalHeader, Tooltip
+} from "@chakra-ui/react"
+import React, { useState } from "react"
+import { CodapModal } from "../codap-modal"
+import t from "../../utilities/translation/translate"
+
+interface IProps {
+  currentValue?: string
+  isOpen: boolean
+  onAccept: (value: string) => void
+  onClose: () => void
+}
+
+export const WebViewUrlModal = ({ currentValue="", isOpen, onAccept, onClose }: IProps) => {
```

Comment:
I notice that pressing <return> doesn't accept value and close the modal. Similarly, <esc> should close without accepting any edits.

Reply:
I was able to address your other three suggestions, but I'm not sure what you mean by this one. Are the buttons of the modal not working for you?

Reply:
Also there's some styling niceties like indicating the OK button is the default and making it disabled when nothing has been entered but enabled when something has been entered that would be nice to observe. That said, I'd also be okay with putting off styling tweaks for a subsequent PR if you'd rather get the basic functionality PR wrapped up first.